src/site/topic16.rst

```diff
@@ -42,6 +42,67 @@ Iterator Interface
 Array Iterator
 --------------
 
+* I do not need to know what kind of underlying container there is for the data to use the iterator
+* However, if I am making my own collection, I will need to create an iterator for that collection and the underlying container
+
+* If the collection we are making is using an array, such as the ``ArrayStack``, we will make an iterator for an array
+    * :download:`ArrayIterator <../src/main/java/ArrayIterator.java>`
```

Comment:
IIRC this class is still untested, only tested indirectly.

Reply:
Good call. Will make new PRs with tests for iterators.